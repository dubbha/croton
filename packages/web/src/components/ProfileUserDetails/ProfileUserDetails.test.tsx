import React from 'react';
import { render } from '@testing-library/react';

describe('components/ProfileUserDetails', () => {
  const setUpMocks = () => {
    jest.doMock('components', () => ({
      Container: () => <div data-testid="CONTAINER" />,
      UpdateProfileForm: () => (
        <form data-testid="UPDATE-PROFILE-FORM">
          <fieldset>
            <input type="text" />
            <input type="submit" />
          </fieldset>
        </form>
      ),
      AddSocialToProfile: () => (
        <button data-testid="WOOSH-WOOSH-AND-SOCIAL-PROFILE-IS-ADDED" />
      ),
    }));

    const { ProfileUserDetails } = require('./ProfileUserDetails');

    return render(<ProfileUserDetails />);
  };

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render successfully', () => {
    jest.isolateModules(() => {
      const { container } = setUpMocks();

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
