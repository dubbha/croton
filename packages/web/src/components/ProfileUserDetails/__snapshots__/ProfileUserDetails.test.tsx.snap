// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ProfileUserDetails should render successfully 1`] = `
<div
  data-testid="CONTAINER"
/>
`;
